src/deepforest/predict.py

```diff
@@ -169,36 +180,40 @@ def _dataloader_wrapper_(model,
         for images in batch:
             prediction_list.append(images)
 
-    results = []
-    for index, prediction in enumerate(prediction_list):
-        # If there is more than one class, apply NMS Loop through images and apply cross
-        if len(prediction.label.unique()) > 1:
-            prediction = across_class_nms(prediction, iou_threshold=nms_thresh)
-
-        prediction["image_path"] = paths[index]
-        results.append(prediction)
+    # Postprocess predictions
+    results = dataloader.dataset.postprocess(prediction_list)
 
-    results = pd.concat(results, ignore_index=True)
     if results.empty:
-        results["geometry"] = None
         return results
 
-    results = read_file(results, root_dir)
+    # Apply across class NMS for each image
+    processed_results = []
+    for image_path in results.image_path.unique():
+        image_results = results[results.image_path == image_path].copy()
 
-    if savedir:
-        visualize.plot_prediction_dataframe(results,
-                                            root_dir=root_dir,
-                                            savedir=savedir,
-                                            color=color,
-                                            thickness=thickness)
+        if crop_model:
+            is_single_model = len(
+                crop_model) == 1  # Flag to check if only one model is passed
```

Comment:
Feels like putting the comment first on its own line so the code can go on a single line would be more readable.